xbmc/filesystem/AddonsDirectory.cpp

```diff
@@ -144,16 +138,11 @@ static bool IsGameAddon(const AddonPtr& addon)
          IsGameSupportAddon(addon);
 }
 
-static bool IsDependencyType(TYPE type)
-{
-  return dependencyTypes.find(type) != dependencyTypes.end();
-}
-
 static bool IsUserInstalled(const AddonPtr& addon)
 {
-  return std::find_if(dependencyTypes.begin(), dependencyTypes.end(), [&](TYPE type) {
+  return std::find_if(ADDON_DEPENDENCY_TYPES.begin(), ADDON_DEPENDENCY_TYPES.end(), [&](TYPE type) {
            return addon->MainType() == type;
-         }) == dependencyTypes.end();
+         }) == ADDON_DEPENDENCY_TYPES.end();
```

Comment:
I would have this call a utility function in CAddonType so that ADDON_DEPENDENCY_TYPES stays encapsulated.

Reply:
I think this can be re-written to use IsDependencyType internally. I'd like not to move this to `CAddonTypes` as its meaning has nothing to do with types

Reply:
That would be even better.

Reply:
How is this different to just calling `!IsDependencyType()`? It not true that an addon is user installed if its not a dependency.

Reply:
It's pretty much the same, the only difference being the fact it receives an addon without specifying what `IsUserInstalled` means. So, it can be changed in the future easily if the meaning also changes.
It is called in a few places, including some lambdas:

https://github.com/xbmc/xbmc/blob/3437652708798220d7f1bb04be79ca4f47aad01e/xbmc/filesystem/AddonsDirectory.cpp#L455

I'd prefer to keep it simple and don't touch many places in this PR. It can be improved later to completely remove this function if preferred. For now, I'll just rewrite it to call the static `IsDependencyType` internally and try to solve the issue this PR is intended at. Otherwise we never finish anything as opportunities for refactoring are plenty :)

Reply:
Ya, I think replacing `std::find_if` with !IsDependencyType(addon->MainType())` makes the most sense.